pkg/kube/diff.go

```diff
@@ -107,6 +107,67 @@ func renderObj(obj runtime.Object, gvk *schema.GroupVersionKind, renderYaml bool
 	return string(yamlBytes), nil
 }
 
+// removeSpuriousDiff implements conditional field removal from live and/or head
+// object depending on the value of the field and the difference between live
+// and head. This behavior differs from that of using the
+// diffFilters, which is unconditional removal.
+//
+// Specifically, this function removes:
+// * Master-assigned NodePort on Live object if head does not specify a NodePort.
+// * Kubernetes namespace finalizer on live and head objects.
+func removeSpuriousDiff(live, head runtime.Object) (newLive, newHead runtime.Object) {
+	newLive, newHead = live.DeepCopyObject(), head.DeepCopyObject()
+	removeSpuriousNodePortDiff(&newLive, &newHead)
+	removeSpuriousNamespaceFinalizerDiff(&newLive, &newHead)
+	return
+}
+
+func removeSpuriousNodePortDiff(live, head *runtime.Object) {
+	liveSvc, ok1 := (*live).(*corev1.Service)
+	headSvc, ok2 := (*head).(*corev1.Service)
+	if !ok1 || !ok2 {
+		return
+	}
+
+	for i, livePort := range liveSvc.Spec.Ports {
+		for _, headPort := range headSvc.Spec.Ports {
+			if livePort.Name == headPort.Name && livePort.NodePort != 0 && headPort.NodePort == 0 {
```

Comment:
do we still need to proceed when both the live and head port name matches? I feel like something like this would be more meaningful. 
```
if livePort.Name == headPort.Name {
    if livePort.NodePort != 0 && headPort.NodePort == 0 {
        liveSvc.Spec.Ports[i].NodePort = 0
    }
    break
}
```

Reply:
Good catch. done.